### PR TITLE
resolves #176 route messages through the logger

### LINKED ISSUE #176: Use logger to report warnings and errors
Use the logger introduced in Asciidoctor 1.5.7 to log warnings and errors.

Note that this will require the minimum version of Asciidoctor to be raised to 1.5.7.

### LINKED ISSUE #196: Example doesn't produce a valid ebook
I saw the travis command running `bundle exec asciidoctor-epub3 -D . data/samples/sample-book.adoc`. It's not validating that the epub is valid. If I run it with `-a ebook-validate ` I can see some errors
```
$ bundle exec asciidoctor-epub3 -D . -a ebook-validate data/samples/sample-book.adoc
asciidoctor: WARNING: asciidoctor-epub3-readme.adoc: invalid reference to anchor in unknown chapter: NOTICE
asciidoctor: WARNING: asciidoctor-epub3-readme.adoc: invalid reference to anchor in unknown chapter: LICENSE
asciidoctor: WARNING: asciidoctor-epub3-readme.adoc: invalid reference to anchor in unknown chapter: NOTICE
Epubcheck Version 3.0.1

Validating against EPUB version 3.0
ERROR: /Users/admejiar/Code/tmp/asciidoctor-epub3/sample-book.epub/OEBPS/asciidoctor-epub3-a-native-epub3-converter-for-asciidoc.xhtml(784,68): 'OEBPS/NOTICE.xhtml': referenced resource missing in the package.
ERROR: /Users/admejiar/Code/tmp/asciidoctor-epub3/sample-book.epub/OEBPS/asciidoctor-epub3-a-native-epub3-converter-for-asciidoc.xhtml(1129,115): 'OEBPS/LICENSE.xhtml': referenced resource missing in the package.
ERROR: /Users/admejiar/Code/tmp/asciidoctor-epub3/sample-book.epub/OEBPS/asciidoctor-epub3-a-native-epub3-converter-for-asciidoc.xhtml(1130,50): 'OEBPS/NOTICE.xhtml': referenced resource missing in the package.

Check finished with warnings or errors
```

### PR DESCRIPTION
**Pull request by slonopotamus**

### CODE REVIEW

**Comment by mojavelinux** [comment]
Lookin' good!

**Comment by slonopotamus** [comment]
@mojavelinux I believe this is ready to be reviewed.

File: bin/asciidoctor-epub3 [hunk 1]
```diff
@@ -14,12 +14,7 @@ options = Asciidoctor::Cli::Options.new backend: 'epub3'
 unless ARGV != ['-v'] && (ARGV & ['-V', '--version']).empty?
   require_relative '../lib/asciidoctor-epub3/version'
   $stdout.write %(Asciidoctor EPUB3 #{Asciidoctor::Epub3::VERSION} using )
-  # NOTE the print_version method was added in Asciidoctor 1.5.2
-  if options.respond_to? :print_version
-    options.print_version
-  else
-    puts %(Asciidoctor #{::Asciidoctor::VERSION} [http://asciidoctor.org])
-  end
+  options.print_version
```
Comment (mojavelinux) [inline bin/asciidoctor-epub3]: I think this change should be made in a separate PR.

File: lib/asciidoctor-epub3/converter.rb [hunk 2]
```diff
@@ -9,6 +9,7 @@ module Epub3
     # EPUB3 or KF8 publication file.
     class Converter
       include ::Asciidoctor::Converter
+      include ::Asciidoctor::Logging
```
Comment (mojavelinux) [inline lib/asciidoctor-epub3/converter.rb]: Interesting idea to encapsulate the shim. I should do that in Asciidoctor PDF. (No need to reply).

File: bin/asciidoctor-epub3 (hunk 1 above)
Comment (mojavelinux) [inline bin/asciidoctor-epub3]: I recommend making this change in a separate PR.

Comment (slonopotamus) [reply bin/asciidoctor-epub3]: Agreed, split into #246.

Comment (slonopotamus) [reply bin/asciidoctor-epub3]: Agreed, split into #246.

**Pull request merged**